plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/gsub/GsubProcessor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.gsub;
+
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.ingest.processor.ConfigurationUtils;
+import org.elasticsearch.ingest.processor.Processor;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class GsubProcessor implements Processor {
+
+    public static final String TYPE = "gsub";
+
+    private final List<GsubExpression> gsubExpressions;
+
+    GsubProcessor(List<GsubExpression> gsubExpressions) {
+        this.gsubExpressions = gsubExpressions;
+    }
+
+    List<GsubExpression> getGsubExpressions() {
+        return gsubExpressions;
+    }
+
+    @Override
+    public void execute(Data data) {
+        for (GsubExpression gsubExpression : gsubExpressions) {
+            String oldVal = data.getPropertyValue(gsubExpression.getFieldName(), String.class);
+            if (oldVal == null) {
+                throw new IllegalArgumentException("field [" + gsubExpression.getFieldName() + "] is null, cannot match pattern.");
+            }
+            Matcher matcher = gsubExpression.getPattern().matcher(oldVal);
+            String newVal = matcher.replaceAll(gsubExpression.getReplacement());
+            data.setPropertyValue(gsubExpression.getFieldName(), newVal);
+        }
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static class Factory implements Processor.Factory<GsubProcessor> {
+        @Override
+        public GsubProcessor create(Map<String, Object> config) throws IOException {
+            List<Map<String, String>> gsubConfig = ConfigurationUtils.readList(config, "expressions");
+            List<GsubExpression> gsubExpressions = new ArrayList<>();
+            for (Map<String, String> stringObjectMap : gsubConfig) {
+                String field = stringObjectMap.get("field");
```

Comment:
should we maybe make ConfigurationUtils more flexible to enable its use here? I understand that the error message may want to be customized in this case.